core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -170,6 +170,13 @@ class KafkaApis(val requestChannel: RequestChannel,
     val response =
       if (authorize(request.session, ClusterAction, Resource.ClusterResource)) {
         val (result, error) = replicaManager.stopReplicas(stopReplicaRequest)
+        result.map{case (topicPartition, errorCode) => {
```

Comment:
Right, the part of the code that you are hinting at will get executed only if the old leader for the __consumer_offsets topic receives a LeaderAndIsr request to become a follower. But in this case it does not receive any such request. 
The only LeaderAndIsr request it receives before it gets the stopReplica request is due to Leader epoch update by the controller,  in which no change to existing Leader or existing ISR is made.
Due to this, the call back "onLeadershipChange" which executes the code to get rid of the information about the particular consumer group from its cache on the old coordinator will never execute and this causes the exception listed in the jira by @jjkoshy